src/main/java/com/commercetools/pspadapter/notification/webhook/impl/WebhookContainerImpl.java

```diff
@@ -22,6 +23,7 @@
 import static java.util.stream.Collectors.toMap;
 
 @Component
+@Profile("default") // skip from integration tests
```

Comment:
Why only this class has this `@Profile` annotation?

Reply:
Because only this one is excluded from the integration tests always.

Reply:
But what happens if you don't have this `@Profile` annotation? I thought you have already `@PrimaryForTest` in the testing bean, so it should be enough?

Reply:
correct, BUT the bean will be created in any case! and `init()` will be called -> execute webhooks requests.

`@PrimaryForTest` indicates this bean should be preferred in favor of other in case of multiple beans exist, but it doesn't prevent bean from creating and calling `init()` inside constructor.

(BTW, this is an evidence of a bad design: this class is not _lazy_ and calls heavy methods even if not required)

Reply:
tried to improve the comment: 2d54afd

Reply:
Maybe it makes sense to create an improvement issue where we can think of how to refactor this class to be lazily initialised and then we won't need the `@Profile` annotation any more as well, as this misuse (since there are no profile and there is no where the active profile is being set. At least that I can see.) of it could be confusing in the future.

Reply:
something about this class that I noticed (not related to this pr), why is the `tenantNameToWebhookCompletionStage` instance variable a `CompletableFuture<Map<String, Webhook>>` and not just a `CompletableFuture<Set<WebhookWithTenantName>>` since I see an [implementation](https://github.com/commercetools/commercetools-paypal-plus-integration/blob/d53415d3e9dd944a53c88b3be378e788aaeb392d/src/main/java/com/commercetools/pspadapter/notification/webhook/impl/WebhookContainerImpl.java#L64-L81) for such a composite object already?

Reply:
To reply to myself about: 

> there is no where the active profile is being set

I see it here now https://github.com/commercetools/commercetools-paypal-plus-integration/pull/130/files/d53415d3e9dd944a53c88b3be378e788aaeb392d#diff-3998c9be5abb9272bc274409cc5df255R4

But still, I think an issue would be useful here to refactor it to lazy init.

Reply:
@heshamMassoud  @lojzatran  #133